omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/UserAppPageController.java

```diff
@@ -52,27 +53,28 @@ public void get(PageModel model, @RequestParam(value = "appId", required = false
 	public String post(PageModel model, @ModelAttribute(value = "appId") @BindParams UserApp userApp,
 	                   @RequestParam("action") String action,
 	                   @SpringBean("appFrameworkService") AppFrameworkService service, HttpSession session, UiUtils ui) {
-		
+
+		String htmlSafeAppId = StringEscapeUtils.escapeHtml(userApp.getAppId());
 		try {
 			AppDescriptor descriptor = mapper.readValue(userApp.getJson(), AppDescriptor.class);
-			if (!userApp.getAppId().equals(descriptor.getId())) {
+			if (!htmlSafeAppId.equals(descriptor.getId())) {
```

Comment:
Is using the `htmlSafeAppId` here instead of the real appId strictly necessary to prevent XSS? (also on line 63)

If not, let's just use the real unescaped appId and only use the html-escaped app-id when it gets displayed in UI messages.

Reply:
@isears I have restricted encoding of appId to just when error messages are displayed